logos/src/logos/capacity/capacity_planner.py

```
@@ -1914,10 +2501,43 @@ async def _ensure_request_capacity(
+        try:
+            await asyncio.wait_for(lock.acquire(), timeout=remaining_for_lock)
+        except asyncio.TimeoutError:
+            logger.info(
+                "ensure_capacity provider=%s model=%s: timed out waiting for "
+                "provider capacity lock (%.1fs)",
+                provider_id, target.model_name, remaining_for_lock,
+            )
+            return False
+
+        try:
+          while True:
```

Flake8 (7.3.0)

[error] indentation is not a multiple of 4

(E111)